metadata-db-app/src/cmr/metadata_db/services/generic_documents.clj

```diff
@@ -26,24 +27,45 @@
                      (dissoc :revision-id))]
     {:concept-id concept-id :provider-id provider-id :revision-id revision-id :metadata metadata}))
 
+;; TODO: Generic work: We need to be able to handle multiple revisions - right now every revision is 1 
```

Comment:
my thinking was to base this on put/post, at least on the metadata layer. I think the more user-friendly behavior belongs in ingest but the database should be explicit.